src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -625,7 +626,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
     -->
   <Target Name="_ComputeNETCoreBuildOutputFiles"
-          DependsOnTargets="_GetAppHostPaths;_GetComHostPaths;_GetIjwHostPaths"
+          DependsOnTargets="_ChooseAppHost;_GetComHostPaths;_GetIjwHostPaths"
```

Comment:
hmm... how that fix the incremental build issue?

Reply:
The root problem here is that ChooseAppHost picks either the app host or the single file host, depending on options, and copies one to the intermediate output directory as 'app.exe'. If you switch which app host you are using by changing the options, it still looks like the copy in the intermediate output directory is up to date because it has the same name. 

With this change we now look for app host.exe or singlefilehost.exe, so if you switch the property you'll look for a different file, and the up to date check will work properly. This relies on publish cleaning the final output directory, but that seems to work fine.